python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -240,24 +248,22 @@ def __init__(
             schema = pa.schema(
                 [schema.field(column) for column in columns], schema.metadata
             )
+            read_schema = schema
 
         check_for_legacy_tensor_type(schema)
 
         if _block_udf is not None:
             # Try to infer dataset schema by passing dummy table through UDF.
             dummy_table = schema.empty_table()
             try:
-                inferred_schema = _block_udf(dummy_table).schema
-                inferred_schema = inferred_schema.with_metadata(schema.metadata)
+                schema = _block_udf(dummy_table).schema
```

Comment:
by overwriting the existing value for `schema` here, would we be losing `schema.metadata` which is used in the next line?

Reply:
oh, nice catch